addons/hr_fleet/models/employee.py

```diff
@@ -75,6 +75,15 @@ def write(self, vals):
             vehicles._compute_mobility_card()
         return res
 
+    def _sync_user(self, user, employee_has_image=False):
+        cars = self.env['fleet.vehicle'].search(['|', ('future_driver_id', '=', self.work_contact_id.id), ('driver_id', '=', self.work_contact_id.id)])
+        for car in cars:
+            if car.future_driver_id == self.work_contact_id:
+                car.future_driver_id = user.partner_id
+            if car.driver_id == self.work_contact_id:
+                car.driver_id = user.partner_id
+        return super()._sync_user(user, employee_has_image=False)
```

Comment:
Let's not override employee_has_image argument
```suggestion
        return super()._sync_user(user, employee_has_image)
```

Reply:
Hello @sofiagvaladze 
I've added the test case :innocent: 
Thank you